newrelic/newrelic.go

```diff
@@ -129,3 +129,39 @@ func ConfigBaseURL(url string) ConfigOption {
 		return errors.New("base URL can not be empty")
 	}
 }
+
+// ConfigInfrastructureBaseURL sets the base URL used to make requests to the Infrastructure API.
+func ConfigInfrastructureBaseURL(url string) ConfigOption {
+	return func(cfg *config.Config) error {
+		if url != "" {
+			cfg.InfrastructureBaseURL = url
+			return nil
+		}
+
+		return errors.New("Infrastructure base URL can not be empty")
```

Comment:
Looking at the provider, if these are not set by the user they would default to an empty string.  While it's a bit awkward, we should accept an empty string and ignore it.  This would be a good logging item to consider.

Reply:
👍